test/TestProjects/PetStore-Typespec/Generated/PetStoreClient.cs

```diff
@@ -326,49 +312,32 @@ public virtual Response GetPetByKind(string kind, RequestContext context = null)
         {
-            using var scope = ClientDiagnostics.CreateScope("PetStoreClient.GetFirstPetValue");
-            scope.Start();
-            try
-            {
-                RequestContext context = FromCancellationToken(cancellationToken);
-                Response response = await GetFirstPetAsync(start, context).ConfigureAwait(false);
-                return Response.FromValue(Pet.FromResponse(response), response);
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
+            RequestContext context = FromCancellationToken(cancellationToken);
+            Response response = await GetFirstPetAsync(start, context).ConfigureAwait(false);
+            return Response.FromValue(Pet.FromResponse(response), response);
         }
 
         /// <param name="start"> The Int32 to use. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual Response<Pet> GetFirstPetValue(int? start = null, CancellationToken cancellationToken = default)
+        public virtual Response<Pet> GetFirstPet(int? start = null, CancellationToken cancellationToken = default)
         {
-            using var scope = ClientDiagnostics.CreateScope("PetStoreClient.GetFirstPetValue");
-            scope.Start();
-            try
-            {
-                RequestContext context = FromCancellationToken(cancellationToken);
-                Response response = GetFirstPet(start, context);
-                return Response.FromValue(Pet.FromResponse(response), response);
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
+            RequestContext context = FromCancellationToken(cancellationToken);
+            Response response = GetFirstPet(start, context);
+            return Response.FromValue(Pet.FromResponse(response), response);
         }
 
         /// <param name="start"> The Int32 to use. </param>
         /// <param name="context"> The request context, which can override default behaviors of the client pipeline on a per-call basis. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="context"/> is null. </exception>
         /// <exception cref="RequestFailedException"> Service returned a non-success status code. </exception>
         /// <returns> The response returned from the service. Details of the response body schema are in the Remarks section below. </returns>
         /// <include file="Docs/PetStoreClient.xml" path="doc/members/member[@name='GetFirstPetAsync(Int32,RequestContext)']/*" />
-        public virtual async Task<Response> GetFirstPetAsync(int? start = null, RequestContext context = null)
+        public virtual async Task<Response> GetFirstPetAsync(int? start, RequestContext context)
         {
+            Argument.AssertNotNull(context, nameof(context));
```

Comment:
I think we do not need to validate context is not null - it is working perfectly fine when we are not validating it.
We are making it required just because the compiler will not accept the overload, but null value is totally acceptable.